specification/sections/05-events.md

```diff
@@ -137,3 +137,21 @@ See [provider initialization](./02-providers.md#24-initialization) and [setting
 > `PROVIDER_READY` handlers attached after the provider is already in a ready state **MUST** run immediately.
 
 See [provider initialization](./02-providers.md#24-initialization) and [setting a provider](./01-flag-evaluation.md#setting-a-provider).
+
+#### Requirement 5.3.4
+
+> If the provider's `on context changed` executes with mismatched contexts, `PROVIDER_STALE` handlers **MUST** run.
+
+See: [provider events](#51-provider-events), [`provider event types`](../types.md#provider-events)
+
+#### Requirement 5.3.5
+
+> If the provider's `on context changed` function successfully updates the cache, `PROVIDER_READY` handlers **MUST** run.
+
+See: [provider events](#51-provider-events), [`provider event types`](../types.md#provider-events)
+
+#### Requirement 5.3.6
+
+> If the provider's `on context changed` function terminates abnormally, `PROVIDER_ERROR` handlers **MUST** run.
+
+See: [provider events](#51-provider-events), [`provider event types`](../types.md#provider-events)
```

Comment:
I like this idea - I think that it could prove useful.

I think one thing I'm unsure about is how to implement `5.3.5`. It would be nice to have these events emitted by the SDK itself, but then I think to satisfy `5.3.5` we would have to _define_ what constitutes a changed context (perhaps a hash of the whole object?). So far, we haven't done this - it's up for the provider to decide what properties they care about on the context, and to decide to take action or not.

I'm hesitant to do this... but maybe it's a good idea for the client?

The other option would be to be less prescriptive here, and simple recommend that provided emit these events themselves when the context has been changed according to their definition.

Reply:
> I think one thing I'm unsure about is how to implement 5.3.5. It would be nice to have these events emitted by the SDK itself, but then I think to satisfy 5.3.5 we would have to define what constitutes a changed context (perhaps a hash of the whole object?). So far, we haven't done this - it's up for the provider to decide what properties they care about on the context, and to decide to take action or not.
I'm hesitant to do this... but maybe it's a good idea for the client?

@toddbaert I am not sure if I get it right but why do we have to be sure that the new context actually is different different influences the flags? To me it fells good enough to say, that a PROVIDER_READY or maybe CONTEXT_CHANGED or so event is dispatched when the context is set and `on context changed` terminated.

Reply:
I think this is a key point, and relates to [this](https://github.com/open-feature/spec/pull/200#discussion_r1275464288).

The language specifically says "mismatched context" which I think means "the context is different from it's previous value". Like I said before, I think this is my biggest concern with this proposal. I agree that it would be easier to simply ignore if the context is different, and run the even regardless. In the client scenario, we expect authors to be mindful of setting the global context... so I think we should just fire the event every time the context is set, regardless of whether it's "different" or not.

Reply:
Yeah, that sounds good to me @toddbaert.

Reply:
> Yeah, I think emitting STALE and READY again even if the context is not mismatched works!

OK great. I think that satisfies this question for me, and we have a consensus. I recommend you update the proposal.

Reply:
Does this means context change (regardless of matching/mismatching ) makes SDK to emit `STALE` status ?

Reply:
My concern is resolved with the latest commit - https://github.com/open-feature/spec/pull/200/commits/8d773c0ef61d4bfe82db1f76d13537f3f8b4ade4